public/src/main/java/com/revenuecat/purchases/models/PurchaseDetails.kt

```diff
@@ -29,11 +28,17 @@ data class PurchaseDetails(
      */
     val signature: String?,
 
-    val originalJson: JSONObject
+    val originalJson: JSONObject,
+
+    val presentedOfferingIdentifier: String?,
+
+    val storeUserID: String?,
+
+    val purchaseType: PurchaseType
 ) : Parcelable
 
-// TODO: should this be nullable or just throw
-val PurchaseDetails.originalGooglePurchase: Purchase?
-    get() = this.signature?.let { signature ->
-        Purchase(this.originalJson.toString(), signature)
-    }
+enum class PurchaseType {
+    GOOGLE_PURCHASE,
+    GOOGLE_RESTORED_PURCHASE,
+    AMAZON_PURCHASE
```

Comment:
Maybe we should call this Amazon_RECEIPT?

Reply:
I kinda prefer `AMAZON_PURCHASE` as a value in an enum called `PurchaseType`